Larger timeout for operations against ProcessGroup for RPC

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38577 larger timeout for operations against ProcessGroup for RPC**

We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump to 2^31 -1. We should bump this to the max value allowed by std::chrono::milliseconds, but in the gloo code this line would then overflow: https://github.com/facebookincubator/gloo/blob/master/gloo/transport/tcp/pair.cc#L896. Filed a issue in gloo to fix this: https://github.com/facebookincubator/gloo/issues/256

closes https://github.com/pytorch/pytorch/issues/33107
closes https://github.com/pytorch/pytorch/issues/38531

Differential Revision: [D21602425](https://our.internmc.facebook.com/intern/diff/D21602425/)

rohan-varma committed:
Infinite timeout for operations against ProcessGroup for RPC

We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump the timeout to the maximum allowed by
std::chrono::milliseconds, which is 2^44 - 1

Differential Revision: [D21602425](https://our.internmc.facebook.com/intern/diff/D21602425/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Infinite timeout for operations against ProcessGroup for RPC"


We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump the timeout to the maximum allowed by
std::chrono::milliseconds, which is 2^44 - 1

closes https://github.com/pytorch/pytorch/issues/33107
closes https://github.com/pytorch/pytorch/issues/38531

Differential Revision: [D21602425](https://our.internmc.facebook.com/intern/diff/D21602425/)

[ghstack-poisoned]

mrshenli commented on torch/distributed/rpc/constants.py:
```diff
@@ -13,7 +12,8 @@
-DEFAULT_PROCESS_GROUP_TIMEOUT = default_pg_timeout
+# Ensure that we don't time out when there are long periods of time without
+# any operations against the underlying ProcessGroup.
+DEFAULT_PROCESS_GROUP_TIMEOUT = timedelta(milliseconds=(2**44) - 1)
```
curious, any reason for using 2^44 - 1 as the default value?

rohan-varma replied:
It's the max of a 45 bit signed integer, because https://en.cppreference.com/w/cpp/chrono/duration says that milliseconds is 45 bits wide. But this has been changed in the latest version of the PR.

mrshenli reviewed (commented):
Thanks for adding this!  Will stamp when failures on `FaultyAgentDistAutogradTestWithSpawn` are gone.

rohan-varma committed:
Update on "Infinite timeout for operations against ProcessGroup for RPC"


We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump the timeout to the maximum allowed by
std::chrono::milliseconds, which is 2^44 - 1

closes https://github.com/pytorch/pytorch/issues/33107
closes https://github.com/pytorch/pytorch/issues/38531

Differential Revision: [D21602425](https://our.internmc.facebook.com/intern/diff/D21602425/)

[ghstack-poisoned]

rohan-varma commented:
@mrshenli Turns out the error was legit, we need to make the timeout lower for now because of this line in gloo causing an overflow: https://github.com/facebookincubator/gloo/blob/master/gloo/transport/tcp/pair.cc#L896. I'll send a diff to fix the overflow possibility in gloo, but would be good to get this in for now i think.

rohan-varma committed:
Update on "Infinite timeout for operations against ProcessGroup for RPC"


We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump to 2^31 -1. We should bump this to the max value allowed by std::chrono::milliseconds, but in the gloo code this line would then overflow: https://github.com/facebookincubator/gloo/blob/master/gloo/transport/tcp/pair.cc#L896

closes https://github.com/pytorch/pytorch/issues/33107
closes https://github.com/pytorch/pytorch/issues/38531

Differential Revision: [D21602425](https://our.internmc.facebook.com/intern/diff/D21602425/)

[ghstack-poisoned]

rohan-varma commented:
@mrshenli, anymore thoughts on this change? Thanks!

mrshenli reviewed (approved):
Thanks for fixing!

rohan-varma committed:
Update on "Infinite timeout for operations against ProcessGroup for RPC"


We don't want to limit a timeout to 30 min since there could be no
operations within that time frame. Bump to 2^31 -1. We should bump this to the max value allowed by std::chrono::milliseconds, but in the gloo code this line would then overflow: https://github.com/facebookincubator/gloo/blob/master/gloo/transport/tcp/pair.cc#L896

closes https://github.com/pytorch/pytorch/issues/33107
closes https://github.com/pytorch/pytorch/issues/38531

Differential Revision: [D21602425](https://our.internmc.facebook.com/intern/diff/D21602425/)

[ghstack-poisoned]
